modules/programs/terminator.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.terminator;
+
+  toValue = val:
+    if val == null then
+      "None"
+    else if val == true then
+      "True"
+    else if val == false then
+      "False"
+    else
+      ''"${toString val}"'';
+
+  toConfigObject = let
+    toKey = depth: key:
+      if depth == 0 then key else toKey (depth - 1) "[${key}]";
+    go = depth: obj:
```

Comment:
`go` is not a very descriptive name for this function. Is there maybe a better way to describe it?

Reply:
Renamed to `toConfigObjectLevel`. `go` is often used in Haskell for recursive helper functions.